Fix Slingshot and Crosshair.

tclaar opened the pull request:
All I did was instantiate those classes, load the crosshair spritesheet, and update some references.

tclaar committed:
Fix Slingshot and Crosshair.
Add crosshair spritesheet to barebones download queue.
Make mouse listeners in GameEngine listen for the correct events.
Update other classes reliant on mouse input to reference new user table members.
Instantiate Slingshot and Crosshair in Zone constructor.

tclaar committed:
Merge pull request #91 from tclaar/prototype

Push Slingshot/Crosshair fix.

calebkrauter commented on js/classes/entities/foreground/crosshair.js:
```diff
@@ -15,8 +15,7 @@ class Crosshair {
         this.pos = GAME.mousePos;
-
-        if (GAME.mouseDown) {
+        if (GAME.user.aiming) {
```
Thanks for doing the refactor here!

NathanHinthorne replied:
Thanks for using the new naming convention

calebkrauter commented on js/classes/game_engine.js:
```diff
@@ -123,14 +123,14 @@ class GameEngine {
-            this.user.aiming = true;
-            this.user.firing = false;
+        CANVAS.addEventListener("mouseup", (e) => {
+            this.user.aiming = false;
```
It's weird to me that everything was reversed prior to your changes.

tclaar replied:
The way you had it made intuitive sense - you were setting firing to true when the mouse was down and setting aiming to true when the mouse was up. It's just that we want our slingshot to work in the opposite way.

calebkrauter commented on js/classes/zone.js:
```diff
@@ -26,6 +26,11 @@ class Zone {
             // Clear all entities from previous Zone.
             GAME.clearEntities();
+
+            // Add entities essential to any Zone.
```
That's a good call.

tclaar replied:
It's possible there is a better place to do this, @B1GD3VD0G would know better than me. It made sense to me to do it here though.

calebkrauter reviewed (approved):
Thanks @tclaar for fixing the slingshot! Not sure if it broke when I did refactoring or prior but it is odd to me that everything was backwards. Regardless the slingshot was not added as an entity when I was working on it. Nice work fixing it up, glad it wasn't too intensive!

tclaar committed:
Add basic combat.
Give Chad health and a takeDamage method.
Make Snake invoke Chad's takeDamage method when attacking.
Make Stone Projectile invoke takeDamage method if applicable when hitting a target.

tclaar committed:
Merge pull request #93 from tclaar/prototype

Add basic combat.

tclaar committed:
Revert "Add basic combat."

tclaar committed:
Merge pull request #94 from GoodBadChad/revert-93-prototype

Revert "Add basic combat."
